Panic when pods have an owner without controllers

```
	owners := p.GetOwnerReferences()
	if len(owners) == 0 {
		addGauge(descPodOwner, 1, "<none>", "<none>", "<none>")
	} else {
		for _, owner := range owners {
			addGauge(descPodOwner, 1, owner.Kind, owner.Name, strconv.FormatBool(*owner.Controller))
		}
	}
```

panics when you have an `Owner` without `Controller`.  `Controller` is an optional field - anyone doing garbage collection (which is what owner is for) will be broken with this check.

```
I0823 14:33:36.780982       1 main.go:243] Starting metrics server: :8080
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x122eb23]

goroutine 245 [running]:
k8s.io/kube-state-metrics/collectors.(*podCollector).collectPod(0xc42040dbe0, 0xc429e3acc0, 0x0, 0x0, 0x0, 0x0, 0xc428c28d40, 0xf, 0x0, 0x0, ...)
	/usr/local/google/home/loburm/go/src/k8s.io/kube-state-metrics/collectors/pod.go:279 +0x313
k8s.io/kube-state-metrics/collectors.(*podCollector).Collect(0xc42040dbe0, 0xc429e3acc0)
	/usr/local/google/home/loburm/go/src/k8s.io/kube-state-metrics/collectors/pod.go:214 +0x117
k8s.io/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func2(0xc427fae8d0, 0xc429e3acc0, 0x1d6dd60, 0xc42040dbe0)
	/usr/local/google/home/loburm/go/src/k8s.io/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus/registry.go:433 +0x61
created by k8s.io/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather
	/usr/local/google/home/loburm/go/src/k8s.io/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus/registry.go:434 +0x27c
```

The Collector dereference should be guarded.

--- a/collectors/pod.go
+++ b/collectors/pod.go
@@ -297,7 +297,11 @@ func (pc *podCollector) collectPod(ch chan<- prometheus.Metric, p v1.Pod) {
 		addGauge(descPodOwner, 1, "<none>", "<none>", "<none>")
 	} else {
 		for _, owner := range owners {
-			addGauge(descPodOwner, 1, owner.Kind, owner.Name, strconv.FormatBool(*owner.Controller))
+			if owner.Controller != nil {
+				addGauge(descPodOwner, 1, owner.Kind, owner.Name, strconv.FormatBool(*owner.Controller))
+			} else {
+				addGauge(descPodOwner, 1, owner.Kind, owner.Name, "false")
+			}
 		}
 	}
 

